### DISCUSSION TITLE
block response whileaudio is played

### DISCUSSION THREAD

**Discussion by jacqueskoreman** [jspsych/jsPsych#1134] [Q&A] [answered]
Hi,

How can I force informants to wait until the audio stimulus (a short dialog) has finished before they press a response button?

Thanks for your reply!
Jacques

**Comment by becky-gilbert** [comment]
Hi @jacqueskoreman, good question. This option should probably be added to the audio plugins as a parameter. Which plugin are you using?

**Reply by becky-gilbert** [reply]
FYI, I suggested this option be added to the audio and video plugins in an issue here: #1137.

**Comment by jacqueskoreman** [comment]
Dear Becky,

I’m using audio-button-response. I attach my code for the experiment (which otherwise does what it is supposed to do).

Thank you for your reply,
Jacques

**Comment by becky-gilbert** [comment, marked as answer]
Hi @jacqueskoreman, as I mentioned in another comment, we're planning to add this option as a plugin parameter in the next release. But in the meantime, you can make this work by modifying your audio-button-response.js plugin file. Specifically, you'll want to (1) disable all of the response buttons when the trial first starts, and (2) enable all of the response buttons once the audio has ended. Here's some code that will do that:
```javascript
    // disable all buttons when the trial starts
    display_element.querySelectorAll('button').forEach(function(btn) {
      btn.disabled = true;
    });

    // enable all the buttons once the audio has ended
    if(context !== null){
      source.addEventListener('ended', function() {
        display_element.querySelectorAll('button').forEach(function(btn) {
          btn.disabled = false;
        });
      });
    } else {
      audio.addEventListener('ended', function() {
        display_element.querySelectorAll('button').forEach(function(btn) {
          btn.disabled = false;
        });
      });
    }
```
This code needs to go in the plugin file _after_ the buttons have been added to the screen, which happens in [this line](https://github.com/jspsych/jsPsych/blob/b8ad672fb7c57f47ddec98ef0a70dca1b72ec78c/plugins/jspsych-audio-button-response.js#L129):
```javascript
display_element.innerHTML = html;
```
And if you're new to JavaScript and/or jsPsych's plugin system, you might be interested in checking out Josh's [YouTube tutorial about creating plugins](https://youtu.be/XQcsFwAmbiw), and the [page on the jsPsych website about creating plugins](https://www.jspsych.org/plugins/creating-a-plugin/).

**Comment by jacqueskoreman** [comment]
Hi!

Perfect! Thank you VERY much for your help and for your very fast repsonse!

Best,
Jacques
